src/modules/iotjs_module_constants.c

```diff
@@ -30,33 +30,192 @@
+#endif
+
+#ifdef SIGKILL
+  NODE_DEFINE_CONSTANT(target, SIGKILL);
+#endif
+
+#ifdef SIGUSR1
+  NODE_DEFINE_CONSTANT(target, SIGUSR1);
+#endif
+
+#ifdef SIGSEGV
+  NODE_DEFINE_CONSTANT(target, SIGSEGV);
+#endif
+
+#ifdef SIGUSR2
+  NODE_DEFINE_CONSTANT(target, SIGUSR2);
+#endif
+
+#ifdef SIGPIPE
+  NODE_DEFINE_CONSTANT(target, SIGPIPE);
+#endif
+
+#ifdef SIGALRM
+  NODE_DEFINE_CONSTANT(target, SIGALRM);
+#endif
+
+  NODE_DEFINE_CONSTANT(target, SIGTERM);
+
+#ifdef SIGCHLD
+  NODE_DEFINE_CONSTANT(target, SIGCHLD);
+#endif
+
+#ifdef SIGSTKFLT
+  NODE_DEFINE_CONSTANT(target, SIGSTKFLT);
+#endif
+
+#ifdef SIGCONT
+  NODE_DEFINE_CONSTANT(target, SIGCONT);
+#endif
+
+#ifdef SIGSTOP
+  NODE_DEFINE_CONSTANT(target, SIGSTOP);
+#endif
+
+#ifdef SIGTSTP
+  NODE_DEFINE_CONSTANT(target, SIGTSTP);
+#endif
+
+#ifdef SIGBREAK
+  NODE_DEFINE_CONSTANT(target, SIGBREAK);
+#endif
+
+#ifdef SIGTTIN
+  NODE_DEFINE_CONSTANT(target, SIGTTIN);
+#endif
+
+#ifdef SIGTTOU
+  NODE_DEFINE_CONSTANT(target, SIGTTOU);
+#endif
+
+#ifdef SIGURG
+  NODE_DEFINE_CONSTANT(target, SIGURG);
+#endif
+
+#ifdef SIGXCPU
+  NODE_DEFINE_CONSTANT(target, SIGXCPU);
+#endif
+
+#ifdef SIGXFSZ
+  NODE_DEFINE_CONSTANT(target, SIGXFSZ);
+#endif
+
+#ifdef SIGVTALRM
+  NODE_DEFINE_CONSTANT(target, SIGVTALRM);
+#endif
+
+#ifdef SIGPROF
+  NODE_DEFINE_CONSTANT(target, SIGPROF);
+#endif
+
+#ifdef SIGWINCH
+  NODE_DEFINE_CONSTANT(target, SIGWINCH);
+#endif
+
+#ifdef SIGIO
+  NODE_DEFINE_CONSTANT(target, SIGIO);
+#endif
+
+#ifdef SIGPOLL
+  NODE_DEFINE_CONSTANT(target, SIGPOLL);
+#endif
+
+#ifdef SIGLOST
+  NODE_DEFINE_CONSTANT(target, SIGLOST);
+#endif
+
+#ifdef SIGPWR
+  NODE_DEFINE_CONSTANT(target, SIGPWR);
+#endif
+
+#ifdef SIGINFO
+  NODE_DEFINE_CONSTANT(target, SIGINFO);
+#endif
+
+#ifdef SIGSYS
+  NODE_DEFINE_CONSTANT(target, SIGSYS);
+#endif
+
+#ifdef SIGUNUSED
+  NODE_DEFINE_CONSTANT(target, SIGUNUSED);
+#endif
+}
+
+void DefinePriorityConstants(jerry_value_t target) {
+  NODE_DEFINE_CONSTANT(target, UV_PRIORITY_LOW);
+  NODE_DEFINE_CONSTANT(target, UV_PRIORITY_HIGHEST);
+}
+
 jerry_value_t InitConstants() {
-  jerry_value_t constants = jerry_create_object();
-
-  SET_CONSTANT(constants, O_APPEND);
-  SET_CONSTANT(constants, O_CREAT);
-  SET_CONSTANT(constants, O_EXCL);
-  SET_CONSTANT(constants, O_RDONLY);
-  SET_CONSTANT(constants, O_RDWR);
-  SET_CONSTANT(constants, O_SYNC);
-  SET_CONSTANT(constants, O_TRUNC);
-  SET_CONSTANT(constants, O_WRONLY);
-  SET_CONSTANT(constants, S_IFMT);
-  SET_CONSTANT(constants, S_IFDIR);
-  SET_CONSTANT(constants, S_IFIFO);
-  SET_CONSTANT(constants, S_IFREG);
-  SET_CONSTANT(constants, S_IFLNK);
-  SET_CONSTANT(constants, S_IFSOCK);
-  SET_CONSTANT(constants, TLS_CHUNK_MAX_SIZE);
-  SET_CONSTANT(constants, INT32_MAX_VALUE);
-  // uv constants
-  SET_CONSTANT(constants, UV_PRIORITY_LOW);
-  SET_CONSTANT(constants, UV_PRIORITY_HIGHEST);
+  jerry_value_t exports = jerry_create_object();
+  jerry_value_t os = jerry_create_object();
```

Comment:
Could this variable be moved to its nearest usage line 217?

Reply:
It makes the first-level objects more clear, we might have more constant groups, if we were to move there as the Latest Usage Rule, that looks like the below:

```js
var exports = {};
//...
var os = {};
exports.os = os;
var signals = {};
os.signals = signals;
//...
var platforms = {};
os.platforms = platforms;

var uv = {};
exports.uv = uv;
var errnos = {};
uv.errnos = errnos;
//...
```
And declare first-level objects firstly looks like:
```js
var exports = {};
var os = {};
var uv = {};
exports.os = os;
exports.uv = uv;
//...
var signals = {};
os.signals = signals;
var platforms = {};
os.platforms = platforms;
//...

var errnos = {};
uv.errnos = errnos;
//...
```
The latter looks more comfortable to me :p